cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -1774,13 +1778,34 @@ pub fn define(
         }
     }
 
+    // for legalizing insertlane with floats, INSERTPS from SSE4.1
+    {
+        let instruction = x86_insertps.bind_vector_from_lane(F32, sse_vector_size);
+        let template = rec_fa_ib.nonrex().opcodes(vec![0x66, 0x0f, 0x3a, 0x21]);
+        e.enc_32_64_maybe_isap(instruction, template, Some(use_sse41_simd));
```

Comment:
We also need encodings when we don't have SSE4.1 here (maybe as a follow-up? Because I just realized that many other vector instructions might be in the same situation).

Reply:
Yeah, I would consider that as a follow-up and perhaps even argue that the effort to add alternate encodings may not be worth the pay-off: how many pre-2006 x86 CPUs will be running cranelift code?

Reply:
Chances are there'll be some of them, if you count in the Firefox user base. I'll try to get data on this. At the moment, support for SSE4.1 support can be enabled or disabled, so if we wanted to make it a strong requirement, we'd need to make this choice more clearer up front (by removing the setting, among other things).

Reply:
Ok, recorded in https://github.com/CraneStation/cranelift/issues/1015.